templates/workspace_services/databricks/terraform/providers.tf

```diff
@@ -0,0 +1,52 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=3.27.0"
+    }
+    azapi = {
+      source  = "Azure/azapi"
+      version = "=1.1.0"
+    }
+    databricks = {
+      source  = "databricks/databricks"
+      version = "1.5.0"
+    }
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {
+    key_vault {
+      # Don't purge on destroy (this would fail due to purge protection being enabled on keyvault)
+      purge_soft_delete_on_destroy               = false
+      purge_soft_deleted_secrets_on_destroy      = false
+      purge_soft_deleted_certificates_on_destroy = false
+      purge_soft_deleted_keys_on_destroy         = false
+      # When recreating an environment, recover any previously soft deleted secrets - set to true by default
+      recover_soft_deleted_key_vaults   = true
+      recover_soft_deleted_secrets      = true
+      recover_soft_deleted_certificates = true
+      recover_soft_deleted_keys         = true
+    }
+  }
+}
+
+provider "azapi" {
+}
+
+provider "databricks" {
+  host                        = azurerm_databricks_workspace.databricks.workspace_url
+  azure_workspace_resource_id = azurerm_databricks_workspace.databricks.id
+
+  azure_use_msi = true
```

Comment:
What does this mean?

Reply:
According to the [docs](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs#authenticating-with-azure-msi) Databricks Terraform provider will use Azure Managed Service Identity to deploy the workspace when ```azure_use_msi``` is set to ```true```.